scavenger-agent-java/src/main/java/com/navercorp/scavenger/javaagent/scheduling/Scheduler.java

```diff
@@ -222,4 +243,28 @@ public void publishInvocationDataIfNeeded() {
             }
         }
     }
+
+    public void publishCallStackDateIfNeeded() {
+        if (config.isCallStackTraceMode() && callStackDataPublisherState.isDueTime() && dynamicConfig != null && isCodeBasePublished) {
```

Comment:
this method is better to just focus on publish. How about checking the CallStackTraceMode on the caller?